### PR TITLE
Add Forgot Password tests

### PR DESCRIPTION
**Pull request by Tiago-Zarzavidjian**
#### :page_facing_up: Description:

This PR creates tests for the forgot password flow.

---

#### :pushpin: Notes:

* The `mockVerifyTokenFailure` replies with `{ errors: null }`, because the endpoint is not returning an error message. We are currently returning an "Invalid token" message if the request fails 
You can check this at: 
https://github.com/loopstudio/react-app-boilerplate/blob/bd1bb16952c877b3d487e488b1f0ee6297250b3b/src/components/ForgotPassword/TokenForm.js#L38
This is being modified in the API, so we'll have to change the response later.

---

#### :heavy_check_mark: Tasks:

* Created tests.
* Added mocks.

---

#### :play_or_pause_button: Demo:

Test coverage improvement:

Before:

image

After:

image

@loopstudio/react-devs

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -48,18 +48,18 @@
     source-map "^0.5.0"
 
 "@babel/core@^7.1.0", "@babel/core@^7.4.5":
-  version "7.11.1"
-  resolved "https://registry.yarnpkg.com/@babel/core/-/core-7.11.1.tgz#2c55b604e73a40dc21b0e52650b11c65cf276643"
-  integrity sha512-XqF7F6FWQdKGGWAzGELL+aCO1p+lRY5Tj5/tbT3St1G8NaH70jhhDIKknIZaDans0OQBG5wRAldROLHSt44BgQ==
+  version "7.11.6"
```
Comment (lucaleivaloop) [inline yarn.lock:51]: Remember to update `package.json` whenever you update `yarn.lock`.

**Review by lucaleivaloop — APPROVED** [review]

**Review by Rminetti — APPROVED** [review]

File: src/components/ForgotPassword/ForgotPassword.test.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  mockResetPasswordFailure,
+} from 'testUtils/mocks/auth';
+import ForgotPassword from '.';
+
+const fakeEmail = 'user@example.com';
+const fakeVerificationCode = '123456';
+const password = 'password';
+
+describe('ForgotPassword', () => {
+  it('should submit correctly', async () => {
```
Comment (mwolman) [inline src/components/ForgotPassword/ForgotPassword.test.js:26]:
```suggestion
  it('submits correctly', async () => {
```

File: src/components/ForgotPassword/ForgotPassword.test.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    const { getByLabelText, getByRole, getByText, history } = renderWithRouter(
+      <ForgotPassword />,
+      {
+        history: ['/forgot-password'],
+      }
```
Comment (mwolman) [inline src/components/ForgotPassword/ForgotPassword.test.js:31]:
```suggestion
      { history: ['/forgot-password'] }
```

File: src/components/ForgotPassword/ForgotPassword.test.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+    fireEvent.click(submitButton);
+
+    await waitFor(() => {
+      expect(mockResetPassword.isDone()).toBeTruthy();
+      expect(history.location.pathname).toEqual('/');
+    });
+  });
+
+  it('should show errors for invalid values', async () => {
```
Comment (mwolman) [inline src/components/ForgotPassword/ForgotPassword.test.js:86]:
```suggestion
  it('shows errors for invalid values', async () => {
```

File: src/components/ForgotPassword/ForgotPassword.test.js [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+    fillInput(email, 'invalid');
+
+    fireEvent.click(submitButton);
+
+    await waitFor(() => {
+      expect(getByText('Invalid email')).toBeInTheDocument();
+    });
+  });
+
+  it('should show error on email form response failure', async () => {
```
Comment (mwolman) [inline src/components/ForgotPassword/ForgotPassword.test.js:101]:
```suggestion
  it('shows error on email form response failure', async () => {
```

File: src/components/ForgotPassword/ForgotPassword.test.js [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+
+    await waitFor(() => {
+      expect(mockedRequest.isDone()).toBeTruthy();
+      expect(
+        getByText(`Unable to find user with email ${fakeEmail}`)
+      ).toBeInTheDocument();
+    });
+  });
+
+  it('should show error on token form response failure', async () => {
```
Comment (mwolman) [inline src/components/ForgotPassword/ForgotPassword.test.js:121]:
```suggestion
  it('shows error on token form response failure', async () => {
```

File: src/components/ForgotPassword/ForgotPassword.test.js [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+
+    fireEvent.click(nextButton);
+
+    await waitFor(() => {
+      expect(mockVerifyToken.isDone()).toBeTruthy();
+      expect(getByText('Invalid token')).toBeInTheDocument();
+    });
+  });
+
+  it('should show error on password form response failure', async () => {
```
Comment (mwolman) [inline src/components/ForgotPassword/ForgotPassword.test.js:155]:
```suggestion
  it('shows error on password form response failure', async () => {
```

**Review by mwolman — APPROVED** [review]

**Review by mconiglio — APPROVED** [review]

**Review by lucaleivaloop — APPROVED** [review]

**Review by ValeFranchi — APPROVED** [review]

**Pull request merged**